source/jormungandr/jormungandr/utils.py

```diff
@@ -460,8 +460,14 @@ class SectionSorter(object):
     def __call__(self, a, b):
         if a.begin_date_time != b.begin_date_time:
             return -1 if a.begin_date_time < b.begin_date_time else 1
-        else:
+        elif a.end_date_time != b.end_date_time:
             return -1 if a.end_date_time < b.end_date_time else 1
+        elif a.destination.uri == b.origin.uri:
+            return -1
+        elif a.origin.uri == b.destination.uri:
+            return 1
+        else:
```

Comment:
I wonder if we should put a log in this `else` case since it's not supposed to happen. Might help to debug in case it actually happens.

Reply:
Indeed, logging (warning or error) here would be nice :+1: